core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -370,6 +383,20 @@ final class DelayedOperationPurgatory[T <: DelayedOperation](purgatoryName: Stri
 
       purged
     }
+
+    def forceComplete(): Int = {
```

Comment:
Actually looking into `Partition.checkEnoughReplicasReachOffset` if we `tryComplete` they  should error out with `Errors.NOT_LEADER_FOR_PARTITION`. This will be returned as `Errors.NOT_COORDINATOR` from the TC. Perhaps that will do?